### ISSUE TITLE
Advanced price export file is empty

### ISSUE THREAD

**Issue by pavru** [magento/magento2#24722]
### Preconditions (*)

1. Magento 2.3.x
2. Create a lot of products more than export page size in my case it's 5000 products.
3. Assign tier price to a product with the lowest ID.

### Steps to reproduce (*)

1. Try to export advanced price.

### Expected result (*)

1. CSV file with one row of tier price should be created.

### Actual result (*)

1. Export error, the file is not created.

### Possible problem.
Probably the bug is in vendor\magento\module-advanced-pricing-import-export\Model\Export\AdvancedPricing.php file.
To find the number of products the collection is loaded with ordering by "has_options" column.
```
            $entityCollection = $this->_getEntityCollection(true);
            $entityCollection->setOrder('has_options', 'asc');
            $entityCollection->setStoreId(Store::DEFAULT_STORE_ID);
            $this->_prepareEntityCollection($entityCollection);
            $this->paginateCollection($page, $this->getItemsPerPage());
            if ($entityCollection->count() == 0) {
                break;
            }
```
Then the same collection is used to get data for exporting with ordering by "entity_id" column **without cleaning**. As a result different set of product is loaded for default store id and the next store id.
```
            $collection->setOrder('entity_id', 'asc');
            $this->_prepareEntityCollection($collection);
            $collection->setStoreId($storeId);
            $collection->load();
            foreach ($collection as $itemId => $item) {
                $data[$itemId][$storeId] = $item;
            }
            $collection->clear();
```

**Comment by pavru** [comment]
Workaround patch:
``` diff
diff --git ../vendor/magento/module-advanced-pricing-import-export/Model/Export/AdvancedPricing.php ../vendor/magento/module-advanced-pricing-import-export/Model/Export/AdvancedPricing.php
index fda6ae95301..04f719dc8d4 100644
--- ../vendor/magento/module-advanced-pricing-import-export/Model/Export/AdvancedPricing.php
+++ ../vendor/magento/module-advanced-pricing-import-export/Model/Export/AdvancedPricing.php
@@ -211,6 +211,7 @@ class AdvancedPricing extends \Magento\CatalogImportExport\Model\Export\Product
             if ($entityCollection->count() == 0) {
                 break;
             }
+            $entityCollection->clear();
             $exportData = $this->getExportData();
             foreach ($exportData as $dataRow) {
                 $writer->writeRow($dataRow);
```

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-20483` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @pavru. Thank you for your report. 
The issue has been fixed in magento/magento2#24831 by @maheshWebkul721 in 2.3-develop branch
Related commit(s):
 - [0c102bb6ed2f5511e364d9bcd38b243343c17a2b](https://github.com/magento/magento2/commit/0c102bb6ed2f5511e364d9bcd38b243343c17a2b)

The fix will be available with the upcoming 2.3.4 release.
